elastalert/alerts.py

```diff
@@ -320,12 +322,22 @@ def __init__(self, rule):
         self.jira_args = {'project': {'key': self.project},
                           'issuetype': {'name': self.issue_type}}
 
+        if self.security_level:
+            self.jira_args['security'] = {'name': self.security_level}
+
         if self.component:
             self.jira_args['components'] = [{'name': self.component}]
         if self.label:
             self.jira_args['labels'] = [self.label]
         if self.assignee:
             self.jira_args['assignee'] = {'name': self.assignee}
+        # Note: because of the sometimes strange behavior of JIRA with custom field types, we have
+        # to format this differently than a text field. Would be nice to allow the user to just
+        # specify the field name and field value and we take care of the rest, but that'd be a lot
+        # of work. For now, we support group picker fields only.
+        if self.custom_group_picker_fields:
+            for key, value in self.custom_group_picker_fields.items():
+                self.jira_args[key] = {'name': value}
```

Comment:
I tried to run with this implementation, but unfortunately it doesn't work for all scenarios.  A custom field (and even built-in fields) can be set in many different ways, depending on the schema of that field:

1. self.jira_args[key] = {'name': 'value'}
2. self.jira_args[key] = [{'name': 'value1'}, {'name': 'value2']
3. self.jira_args[key] = 'value'
4. self.jira_args[key] = ['value1', 'value2']
5. self.jira_args[key] = {'id': 'value'}
6. self.jira_args[key] = [{'id': 'value1'}, {'id': 'value2']
7. self.jira_args[key] = {'key': 'value'}
8. self.jira_args[key] = [{'key': 'value1'}, {'key': 'value2']

And so on potentially...  There needs to be a lot more smarts than blindly taking the custom fields and setting them in the format you've suggested.  Furthermore, it will be preferable if users can express their field names with the public facing names ('Software Users') rather than the internal representation (customfield_1234).

I have an idea for how to make this work for cases 1-4 above, && allow the user to either specify the customfield_1234 OR the friendly name in snake_case (e.g. 'Software Users' would become 'jira_software_users').  I will hope to send a new pull request within the next couple of days.